File: C_Programming/Practise_Programs/PP_069.c
/*
==============================================================================
 File Name   : PP_069.c

 Program     : Traverse Array Elements Using an Array Parameter

 Description : Demonstrates that an array parameter behaves like a pointer
               inside a function. Pointer arithmetic is used to access
               consecutive elements of the array.

 Author      : Shubham Gadhe

 Time Complexity  : O(1)
 Space Complexity : O(1)
==============================================================================
*/

#include<stdio.h>

/*
==============================================================================
 Function Name : Display

 Description   : Displays the first three elements of the array using pointer
                 dereferencing and pointer arithmetic. Although the parameter
                 is declared as an array, it is treated as a pointer inside
                 the function.

 Input         : Integer array
 Output        : None
==============================================================================
*/

void Display(int Arr[])
{
    // Display the first element
    printf("%d\n", *Arr);      
    
    // Move to the next array element
    Arr++;

    // Move to the next array element
    printf("%d\n", *Arr); 

    // Display the third element
    Arr++;

    printf("%d\n", *Arr);
}

/*
==============================================================================
 Entry Point Function
==============================================================================
*/

int main()
{
   int Brr[5] = {10,20,30,40,50};  

   // Pass the array to the function
   Display(Brr);    

   return 0;  
}

/*
==============================================================================
 Sample Output
------------------------------------------------------------------------------
10
20
30
==============================================================================
*/

File: Java_Programming/Practise_Programs/PP_069.java
//////////////////////////////////////////////////////////////////
//
//  File Name         : PP_069
//
//  Description       : This program accepts the number of rows
//                      and columns from the user and displays
//                      '$' in odd rows and '#' in even rows
//                      using a character array.
//
//  Author            : Shubham Somanath Gadhe
//  Date              : 08/08/2026
//
//  Time Complexity   : O(r × c)
//  Space Complexity  : O(1)
//
//////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Pattern
{
    //////////////////////////////////////////////////////////////////
    //
    //  Function Name    : Display()
    //
    //  Description      : It is used to display '$' in odd
    //                     numbered rows and '#' in even
    //                     numbered rows by selecting the
    //                     character from an array.
    //
    //  Author           : Shubham Somanath Gadhe
    //  Date             : 08/08/2026
    //
    //////////////////////////////////////////////////////////////////

    public void Display(int iRow, int iCol)
    {
        int i = 0, j = 0;
        char Arr[] = {'#', '$'};

        for(i = 1; i <= iRow; i++)
        {
            for(j = 1; j <= iCol; j++)
            {
                System.out.print(Arr[i % 2] + "\t");
            }

            System.out.println();
        }
    }
}

class PP_069
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue1 = 0, iValue2 = 0;

        System.out.println("Enter the number of Rows: ");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the number of Column: ");
        iValue2 = sobj.nextInt();

        Pattern pobj = new Pattern();

        pobj.Display(iValue1, iValue2);
    }
}

//////////////////////////////////////////////////////////////////
//
//  Sample Input  :
//      Enter the number of Rows :
//      4
//
//      Enter the number of Column :
//      4
//
//  Sample Output :
//
//      $    $    $    $
//      #    #    #    #
//      $    $    $    $
//      #    #    #    #
//
//////////////////////////////////////////////////////////////////